src/overlays/actors/ovl_En_Vali/z_en_vali.c

```diff
@@ -76,71 +82,732 @@ static ColliderCylinderInit D_80B288F0 = {
+    this->actionFunc = EnVali_FloatIdle;
+}
+
+/**
+ * Used for both touching player/player's shield and being hit with sword. What to do next is determined by params.
+ */
+void EnVali_SetupAttacked(EnVali* this) {
+    this->lightningTimer = 20;
+    this->actor.flags &= ~1;
+    this->bodyCollider.base.acFlags &= ~AC_ON;
+    this->actionFunc = EnVali_Attacked;
+}
+
+void EnVali_SetupRetaliate(EnVali* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &gBariRetaliatingAnim, -5.0f);
+    Actor_SetColorFilter(&this->actor, 0x4000, 150, 0x2000, 30);
+    this->actor.params = BARI_TYPE_NORMAL;
+    this->bodyCollider.base.acFlags &= ~AC_ON;
+    this->actionFunc = EnVali_Retaliate;
+}
+
+void EnVali_SetupMoveArmsDown(EnVali* this) {
+    Animation_PlayOnce(&this->skelAnime, &gBariMovingArmsDownAnim);
+    this->actionFunc = EnVali_MoveArmsDown;
+}
+
+void EnVali_SetupBurnt(EnVali* this) {
+    this->timer = 2;
+    this->bodyCollider.base.acFlags &= ~AC_ON;
+    Actor_SetColorFilter(&this->actor, 0x4000, 150, 0x2000, 30);
+    this->actionFunc = EnVali_Burnt;
+}
+
+void EnVali_SetupDivideAndDie(EnVali* this, GlobalContext* globalCtx) {
+    s32 i;
+
+    for (i = 0; i < 3; i++) {
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BILI, this->actor.world.pos.x, this->actor.world.pos.y,
+                    this->actor.world.pos.z, 0, this->actor.world.rot.y, 0, 0);
+
+        this->actor.world.rot.y += 0x10000 / 3;
+    }
+
+    Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos, 0x50);
+    this->timer = Rand_S16Offset(10, 10);
+    this->bodyCollider.base.acFlags &= ~AC_ON;
+    Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 40, NA_SE_EN_BARI_SPLIT);
+    this->actor.flags &= ~1;
+    this->actor.draw = NULL;
+    this->actionFunc = EnVali_DivideAndDie;
+}
+
+void EnVali_SetupStunned(EnVali* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &gBariWaitingAnim, 10.0f);
+    this->timer = 80;
+    this->actor.velocity.y = 0.0f;
+    Actor_SetColorFilter(&this->actor, 0, 255, 0x2000, 80);
+    this->bodyCollider.info.bumper.effect = 0;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+    this->actor.velocity.y = 1.0f;
+    this->actionFunc = EnVali_Stunned;
+}
+
+void EnVali_SetupFrozen(EnVali* this) {
+    this->actor.velocity.y = 0.0f;
+    Actor_SetColorFilter(&this->actor, 0, 255, 0x2000, 36);
+    this->bodyCollider.base.acFlags &= ~AC_ON;
+    this->timer = 36;
+    this->actionFunc = EnVali_Frozen;
+}
+
+void EnVali_SetupReturnToLurk(EnVali* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &gBariLurkingAnim, 10.0f);
+    this->actor.flags |= 0x10;
+    this->actor.flags &= ~1;
+    this->actionFunc = EnVali_ReturnToLurk;
+}
+
+void EnVali_DischargeLightning(EnVali* this, GlobalContext* globalCtx) {
+    static Color_RGBA8 effectPrimColor = { 255, 255, 255, 255 };
+    static Color_RGBA8 effectEnvColor = { 200, 255, 255, 255 };
+    Vec3f effectPos;
```

Comment:
in general I prefer to name these after the specific effect, unless its reused for multiple. so i this case its for the shock effect
this is the kinda thing we can fix later on, but just putting it out there

Reply:
So `lightningEffectPos`, or `lightningPos`?

Reply:
tbh in this context since its only spawning one thing, I tend to just call it `pos` `primColor` etc, and only specify if needed. but if i were to specify, id leave out "effect"